# Confirmez vos compétences en MLOps (partie 2/2)

# Imports dépendances

In [1]:
# %pip install -e .

In [2]:
pip list

Package                            Version      Editable project location
---------------------------------- ------------ -------------------------------------------------------------------------------------------
aiohappyeyeballs                   2.7.1
aiohttp                            3.14.3
aiosignal                          1.4.0
alembic                            1.18.5
annotated-doc                      0.0.4
annotated-types                    0.8.0
anyio                              4.14.2
asttokens                          3.0.2
attrs                              26.1.0
blinker                            1.9.0
cachetools                         7.1.6
category_encoders                  2.10.0
certifi                            2026.7.22
cffi                               2.1.0
charset-normalizer                 3.4.9
click                              8.4.2
cloudpickle                        3.1.2
colorama                           0.4.6
comm                               0.2.

In [3]:
import joblib
import os
from pathlib import Path

import pandas as pd

# Extraire le modèle

In [4]:
# # 1. Création du dossier artifacts
# os.makedirs("artifacts", exist_ok=True)

# # 2. Exportation directe du pipeline/modèle
# # (Remplacez 'best_pipeline' par le nom exact de votre variable)
# joblib.dump(new_pipeline_xgb, "artifacts/best_model_pipeline.joblib")

# print("✅ Artefact du modèle sauvegardé avec succès dans artifacts/best_model_pipeline.joblib")

In [5]:
print("📋 Fichiers et dossiers dans le projet actuel :")
for root, dirs, files in os.walk("."):
    # On ignore le dossier de l'environnement virtuel .venv
    if ".venv" in root or ".git" in root:
        continue
    for file in files:
        if file.endswith((".db", ".pkl", ".joblib", ".tar.gz")):
            print(" 📄", os.path.join(root, file))

📋 Fichiers et dossiers dans le projet actuel :
 📄 .\mlflow.db
 📄 .\mlruns\1\models\m-08c1abcedc9d466f8c367563c8e2b046\artifacts\model.pkl
 📄 .\mlruns\1\models\m-26d7e5e547454dc98ed232ee7568f67b\artifacts\model.pkl
 📄 .\mlruns\1\models\m-27e71ad869024a0696aa65649b930005\artifacts\model.pkl
 📄 .\mlruns\1\models\m-2cd607759efb42e8aa6a0dd79b307967\artifacts\model.pkl
 📄 .\mlruns\1\models\m-325a052b8c8f4a309e5b876e4ba3b92e\artifacts\model.pkl
 📄 .\mlruns\1\models\m-4f660221d6f44a27a9ecdf38711665fb\artifacts\model.pkl
 📄 .\mlruns\1\models\m-626fc0d1fbf94c6284d782ef15833f0e\artifacts\model.pkl
 📄 .\mlruns\1\models\m-6e2fde70404c4b62b199dd032a7444a0\artifacts\model.pkl
 📄 .\mlruns\1\models\m-85fd706604a644129a01815d9c093b87\artifacts\model.pkl
 📄 .\mlruns\1\models\m-8f7cf7d301704324b46d69ecc0c8e922\artifacts\model.pkl
 📄 .\mlruns\1\models\m-d00be07eb0494ad382b8361e1c0ef33b\artifacts\model.pkl
 📄 .\mlruns\1\models\m-d2675cd5e56a464d967bd03eb5d3bea4\artifacts\model.pkl
 📄 .\mlruns\1\models\m-d5d

In [6]:
PROJECT_ROOT = Path.cwd().parent if Path.cwd().name == "notebooks" else Path.cwd()

# Chemin vers le modèle
MODEL_PATH = PROJECT_ROOT / "artifacts" / "best_model_pipeline.joblib"

print(f"📍 Chemin du modèle : {MODEL_PATH}")

# Chargement du modèle
pipeline = joblib.load(MODEL_PATH)
print("✅ Modèle chargé avec succès !")

📍 Chemin du modèle : c:\Users\15GIRAV\Moi\Boulot\OpenClassrooms\P8_Confirmez_vos_competences_en_MLOPS_partie_2_2\artifacts\best_model_pipeline.joblib
✅ Modèle chargé avec succès !


Étape 2.1 : Écriture du code de l'API FastAPI (src/main.py + schéma Pydantic) et test local (uvicorn).

Étape 2.2 : Rédaction des tests pytest (cas valides + cas limites/erreurs).

Étape 2.3 : Création et validation du Dockerfile localement (docker build & docker run).

Étape 2.4 : Configuration du Space Hugging Face et du workflow GitHub Actions (ci-cd.yml).